test/TestProjects/Spread-Cadl/Generated/SpreadClient.cs

```diff
@@ -496,6 +498,134 @@ public virtual Response SpreadAliasWithSpreadAlias(string id, int top, RequestCo
             }
         }
 
+        /// <summary> spread an alias with contains optional properties as body. </summary>
+        /// <param name="id"> The String to use. </param>
+        /// <param name="top"> The Int32 to use. </param>
+        /// <param name="name"></param>
+        /// <param name="items"></param>
+        /// <param name="color"></param>
+        /// <param name="age"></param>
+        /// <param name="elements"></param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        /// <exception cref="ArgumentNullException"> <paramref name="id"/>, <paramref name="name"/> or <paramref name="items"/> is null. </exception>
+        /// <exception cref="ArgumentException"> <paramref name="id"/> is an empty string, and was expected to be non-empty. </exception>
+        public virtual async Task<Response> SpreadAliasWithOptionalPropsAsync(string id, int top, string name, IEnumerable<int> items, string color = null, int? age = null, IEnumerable<string> elements = null, CancellationToken cancellationToken = default)
```

Comment:
What is the order of required parameters? i.e. why top is before name?

Reply:
This logic exists before. Order of parameters:
Required path -> Required query -> body -> optional query (we cannot have an optional path therefore it is skipped here)

In this PR, we spread body into a list of parameters, they replace body in the above list.
Therefore
id (required path), top (required query), name (required spread body), items (required spread body), others (optional spread body)